src/main/kotlin/apply/application/ApplicantService.kt

```diff
@@ -0,0 +1,50 @@
+package apply.application
+
+import apply.domain.applicant.Applicant
+import apply.domain.applicant.ApplicantRepository
+import apply.domain.applicant.Gender
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import javax.annotation.PostConstruct
+
+@Transactional
+@Service
+class ApplicantService(val applicantRepository: ApplicantRepository) {
+    fun findAll(): List<Applicant> {
+        return applicantRepository.findAll()
+    }
+
+    fun getByName(name: String): Applicant =
+        applicantRepository.findByName(name).orElse(null)
```

Comment:
`null`이 반환될 수 있기 때문에 `Applicant`를 반환 타입으로 선언하는 것은 위험합니다. 그렇다면 `Applicant?`가 올바른 반환 타입인데 이 경우 함수 이름으로 'get'보다 'find'가 더 적절해 보여요. 아래의 글을 참고하세요.
https://tuhrig.de/find-vs-get

Reply:
현재 `null`이면 `IllegalStateException`이 발생합니다. 코틀린 1.4부터 `NullPointerException`이 발생합니다.

> Starting from Kotlin 1.4.0, all runtime null checks will throw a `java.lang.NullPointerException` instead of `KotlinNullPointerException`, `IllegalStateException`, `IllegalArgumentException`, and `TypeCastException`.

Reply:
네이밍과 예외관련하여 배워갑니다..👍

Reply:
이 부분이 검색할 때 불편하다고 생각되네요!

정확한 이름을 검색하는 것보다 **검색어를 포함**하면 결과를 보여주는 게 더 자연스러울 것 같아요.

그리고 이름보다는 이메일로 검색할 경우도 많을 것 같아서 이름과 이메일 모두로 검색이 가능하면 더 좋을 것 같네요 😄 

또, 검색결과가 여러개면 예외가 발생합니다. `List`로 반환해주는 건 어떨까요??

Reply:
몰랐던 부분인데 내용 감사합니다!